homeassistant/components/vicare/number.py

```diff
@@ -0,0 +1,217 @@
+    """Representation of a ViCare number."""
+
+    entity_description: ViCareNumberEntityDescription
+
+    def __init__(
+        self,
+        name: str,
+        api: PyViCareHeatingDeviceWithComponent,
+        device_config: PyViCareDeviceConfig,
+        description: ViCareNumberEntityDescription,
+    ) -> None:
+        """Initialize the number."""
+        super().__init__(device_config)
+        self.entity_description = description
+        self._attr_name = name
+        self._api = api
+        self._device_config = device_config
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._attr_native_value is not None
+
+    @property
+    def unique_id(self) -> str:
+        """Return unique ID for this device."""
+        tmp_id = (
+            f"{self._device_config.getConfig().serial}-{self.entity_description.key}"
+        )
+        if hasattr(self._api, "id"):
+            return f"{tmp_id}-{self._api.id}"
+        return tmp_id
+
+    async def async_set_native_value(self, value: float) -> None:
+        """Set new value."""
+        if self.entity_description.value_setter:
+            await self.hass.async_add_executor_job(
+                self.entity_description.value_setter, self._api, value
+            )
+        self.async_write_ha_state()
+
+    def update(self) -> None:
+        """Update state of number."""
+        try:
+            with suppress(PyViCareNotSupportedFeatureError):
+                value = self.entity_description.value_getter(self._api)
+                if value is not None:
+                    self._attr_native_value = float(value)
+
+                if self.entity_description.min_value_getter:
+                    min_value = self.entity_description.min_value_getter(self._api)
+                    if min_value is not None:
+                        self._attr_native_min_value = min_value
+
+                if self.entity_description.max_value_getter:
+                    max_value = self.entity_description.max_value_getter(self._api)
+                    if max_value is not None:
+                        self._attr_native_max_value = max_value
+
+                if self.entity_description.stepping_getter:
+                    stepping = self.entity_description.stepping_getter(self._api)
+                    if stepping is not None:
+                        self._attr_native_step = stepping
+        except RequestConnectionError:
+            _LOGGER.error("Unable to retrieve data from ViCare server")
+        except ValueError:
+            _LOGGER.error("Unable to decode data from ViCare server")
+        except PyViCareRateLimitError as limit_exception:
+            _LOGGER.error("Vicare API rate limit exceeded: %s", limit_exception)
+        except PyViCareInvalidDataError as invalid_data_exception:
+            _LOGGER.error("Invalid data from Vicare server: %s", invalid_data_exception)
```

Comment:
Can we do something better here than flooding the log in case there is repeating errors?

Reply:
We probably could raise `PlatformNotReady` instead, but I suggest to handle that as a whole for the other platforms as well.

Reply:
This should probably have a `DataUpdateCoordinator` anyhow but was more thinking can we log once instead of repeating every x seconds when it tries to refresh the data.